### PR TITLE
Add support for `-preview=in`

### PR DESCRIPTION
**Pull request by Geod24**
Just a bunch of sed, nothing crazy.

### CODE REVIEW

File: source/cblas/cblas.di [hunk 1]
```diff
@@ -139,333 +139,333 @@ void cblas_xerbla(blasint p, char *rout, char *form, ...);
 pure:
 
 ///
-float  cblas_sdsdot(in blasint n, in float alpha, in float *x, in blasint incx, in float *y, in blasint incy);
+float  cblas_sdsdot(blasint n, float alpha, float *x, blasint incx, float *y, blasint incy);
```
Comment (9il) [inline source/cblas/cblas.di]: `in` everywhere in mir and in colas means `scope const`, can't be replaced with anything else

Comment (9il) [reply source/cblas/cblas.di]: `in` doesn't make sense for value types, but important for pointers and ref values.

Comment (Geod24) [reply source/cblas/cblas.di]: Okay, will change

**Comment by PetarKirov** [comment]
ping @Geod24

**Comment by Geod24** [comment]
Fixed

**Comment by 9il** [comment]
`in` have to be replaced with `scope const`. A simple replacement. This MR does something different.

**Comment by Geod24** [comment]
I don't follow. What's the issue ?

**Comment by 9il** [comment]
The PR breaks the API. `in` has to be replaced with `scope const`.

File: source/cblas/cblas.di [hunk 2]
```diff
@@ -139,333 +139,333 @@ void cblas_xerbla(blasint p, char *rout, char *form, ...);
...
 
 ///
-void  cblas_cdotu_sub(in blasint n, in _cfloat  *x, in blasint incx, in _cfloat  *y, in blasint incy, _cfloat  *ret);
+void  cblas_cdotu_sub(blasint n, _cfloat  *x, blasint incx, _cfloat  *y, blasint incy, _cfloat  *ret);
 ///
-void  cblas_cdotc_sub(in blasint n, in _cfloat  *x, in blasint incx, in _cfloat  *y, in blasint incy, _cfloat  *ret);
+void  cblas_cdotc_sub(blasint n, _cfloat  *x, blasint incx, _cfloat  *y, blasint incy, _cfloat  *ret);
 ///
-void  cblas_zdotu_sub(in blasint n, in _cdouble *x, in blasint incx, in _cdouble *y, in blasint incy, _cdouble *ret);
+void  cblas_zdotu_sub(blasint n, _cdouble *x, blasint incx, _cdouble *y, blasint incy, _cdouble *ret);
```
Comment (9il) [inline source/cblas/cblas.di]: `_cdouble *x` become mutable. It should be `scope const _cdouble *x`

File: source/cblas/cblas.di [hunk 3]
```diff
@@ -139,333 +139,333 @@ void cblas_xerbla(blasint p, char *rout, char *form, ...);
...
+void cblas_saxpy(blasint n, float alpha, float *x, blasint incx, float *y, blasint incy);
 ///
-void cblas_daxpy(in blasint n, in double alpha, in double *x, in blasint incx, double *y, in blasint incy);
+void cblas_daxpy(blasint n, double alpha, double *x, blasint incx, double *y, blasint incy);
 ///
-void cblas_caxpy(in blasint n, in ref _cfloat alpha, in _cfloat *x, in blasint incx, _cfloat *y, in blasint incy);
+void cblas_caxpy(blasint n, scope const _cfloat* alpha, _cfloat *x, blasint incx, _cfloat *y, blasint incy);
 ///
-void cblas_zaxpy(in blasint n, in ref _cdouble alpha, in _cdouble *x, in blasint incx, _cdouble *y, in blasint incy);
+void cblas_zaxpy(blasint n, scope const _cdouble* alpha, _cdouble *x, blasint incx, _cdouble *y, blasint incy);
```
Comment (9il) [inline source/cblas/cblas.di]: Why `ref` replaced with a pointer?

**Comment by Geod24** [comment]
@9il : Can you re-review ? Reverted the `ref` => pointer changes, it should all be G2G.

**Comment by Geod24** [comment]
Thanks! Can you make a release too, preferably a patch one since the API hasn't changed.

**Pull request merged**